client/src/jobq/image.py

```diff
@@ -1,3 +1,15 @@
+from enum import Enum
+
+
+class ImagePullPolicy(Enum):
+    ALWAYS = "Always"
+    NEVER = "Never"
+    IFNOTPRESENT = "IfNotPresent"
+
+
 class Image:
-    def __init__(self, tag: str) -> None:
+    def __init__(
+        self, tag: str, pull_policy: ImagePullPolicy = ImagePullPolicy.ALWAYS
+    ) -> None:
         self.tag = tag
+        self.pull_policy = pull_policy
```

Comment:
I don't think this is the right place for it.

An image by itself (which this class models) does not have a pull policy

The only place a pull policy matters is when submitting a job (and that's currently not covered by the PR). I would see the `pull_policy` as an addition to `CreateJobModel` (as you have it), and then as part of the signature of `Runner.run`.

As for the actual runner implementations, it's straightforward for the Kubernetes-based modes. For Docker, there's a [`--pull` flag](https://docs.docker.com/reference/cli/docker/container/run/) to `docker run`, which controls the behavior.